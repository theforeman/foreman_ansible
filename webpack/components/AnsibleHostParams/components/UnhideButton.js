import React from 'react';
import { translate as __ } from 'foremanReact/common/I18n';
import PropTypes from 'prop-types';

import { Button } from 'patternfly-react';

import ButtonHelpPopover from './ButtonHelpPopover';

const UnhideButton = ({ toggleHidden, hidden, lookupKey }) => {
  if (!lookupKey.hiddenValue) {
    return '';
  }

  let faIcon;
  let popoverText;
  let popoverId;

  const keyId = lookupKey.id;

  if (hidden) {
    popoverText = __('Unhide this value');
    popoverId = `lookup-key-unhide-value-${keyId}`;
  } else {
    popoverText = __('Hide this value');
    popoverId = `lookup-key-hide-value-${keyId}`;
    faIcon = 'btn-strike';
  }

  const button = (
    <Button
      name="button"
      type="button"
      className="btn btn-default btn-md btn-hide"
      onClick={toggleHidden}
    >
      <span className={`fa fa-font ${faIcon}`} />
    </Button>
  );

  return (
    <ButtonHelpPopover
      button={button}
      popoverText={popoverText}
      popoverId={popoverId}
    />
  );
};

UnhideButton.propTypes = {
  toggleHidden: PropTypes.func.isRequired,
  hidden: PropTypes.bool.isRequired,
  lookupKey: PropTypes.object.isRequired,
};

export default UnhideButton;
