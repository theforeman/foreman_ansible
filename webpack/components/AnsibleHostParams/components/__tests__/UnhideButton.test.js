import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import UnhideButton from '../UnhideButton';

const toggleHidden = () => {};

const fixtures = {
  'should render when value is not hidden': {
    lookupKey: {
      hiddenValue: 'foo',
      id: 80,
    },
    hidden: false,
    toggleHidden,
  },
  'should render when value is hidden': {
    lookupKey: {
      hiddenValue: 'foo',
      id: 80,
    },
    hidden: false,
    toggleHidden,
  },
  'should not show when not needed': {
    lookupKey: {},
    toggleHidden,
    hidden: false,
  },
};

describe('UnhideButton', () =>
  testComponentSnapshotsWithFixtures(UnhideButton, fixtures));
