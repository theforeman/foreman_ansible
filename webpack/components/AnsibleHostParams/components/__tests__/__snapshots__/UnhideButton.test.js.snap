// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnhideButton should not show when not needed 1`] = `""`;

exports[`UnhideButton should render when value is hidden 1`] = `
<ButtonHelpPopover
  button={
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="btn btn-default btn-md btn-hide"
      disabled={false}
      name="button"
      onClick={[Function]}
      type="button"
    >
      <span
        className="fa fa-font btn-strike"
      />
    </Button>
  }
  popoverId="lookup-key-hide-value-80"
  popoverText="Hide this value"
/>
`;

exports[`UnhideButton should render when value is not hidden 1`] = `
<ButtonHelpPopover
  button={
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="btn btn-default btn-md btn-hide"
      disabled={false}
      name="button"
      onClick={[Function]}
      type="button"
    >
      <span
        className="fa fa-font btn-strike"
      />
    </Button>
  }
  popoverId="lookup-key-hide-value-80"
  popoverText="Hide this value"
/>
`;
